datahandling.py

```diff
@@ -340,8 +338,12 @@ def store_bodyfetcher_max_ids():
     _dump_pickle("bodyfetcherMaxIds.p", GlobalVars.bodyfetcher.previous_max_ids)
 
 
-def store_queue_timings():
-    _dump_pickle("bodyfetcherQueueTimings.p", GlobalVars.bodyfetcher.queue_timings)
+def add_queue_timing_data(site, time_in_queue):
+    with open("bodyfetcherQueueTimings.txt", mode="a", encoding="utf-8") as stat_file:
```

Comment:
Can't we just open the file once on startup, instead of every time we write to it?

Reply:
We could, but that would result in Smokey holding exclusive access to the file while it's running, whereas opening on demand leaves the file available - for instance if you want to edit/correct it while Smokey is running.

Reply:
To open the file on startup means that we need to somehow close it when SD shuts down. Maybe we can use some hooks but I am not sure how to do it.

---
> but that would result in Smokey holding exclusive access to the file while it's running

It appears that (at least under linux/bsd) it is not the case, and we are not even using POSIX advisory lock. (And we can ignore the advisory lock anyway since it is not a mandatory lock.)

Reply:
Aren't open files closed automatically when the process exits?

Reply:
How about we hold this data in RAM for a relatively short while and just write once every N times that `add_queue_timing_data()` is called, or if M time has passed since the last time which it was called (and upon SD exit)? Do we _need_ this data to never have lost some data points (i.e. do we need to be _certain_ we don't loose any data if SD hard crashes)?

Reply:
That seems reasonable to me -- it sure would save a lot of disk traffic.

I don't think we care too much about preserving every data point. AFAIK all we ever do with this data is average it all up (and use the results to periodically adjust queue sizes), so dropped data points shouldn't have a significant impact.

Reply:
> Aren't open files closed automatically when the process exits?

Well, (I think) data integrity is not guaranteed in such cases.

Reply:
> How about we hold this data in RAM for a relatively short while and just write once every N times that `add_queue_timing_data()` is called, or if M time has passed since the last time which it was called (and upon SD exit)? Do we _need_ this data to never have lost some data points (i.e. do we need to be _certain_ we don't loose any data if SD hard crashes)?

Do we really need that much extra control structure/logic for negligible savings in total disk I/O?

Reply:
> Well, (I think) data integrity is not guaranteed in such cases.

Aw, looks like you're right about that. The OS guarantees data integrity, but the Python runtime does its own buffering and because we use `os._exit` instead of `sys.exit`, Python can't guarantee those buffers get flushed.

> Do we really need that much extra control structure/logic for negligible savings in total disk I/O?

Well, the purpose of this PR is to reduce disk I/O after all. It's possible that those changes would be unnecessary because of the OS's own caching/buffering, but I'm not sure how much we can rely on that. Opening a file, writing a single line, and closing a file for *every post we scan* is a lot of work.